contracts/examples/Collateralized.sol

```diff
@@ -60,158 +70,309 @@ contract Collateralized is TermsContract {
     );
 
-    function Collateralized(address _debtRegistry) public {
+    function Collateralized(address _debtKernel, address _debtRegistry, address _tokenRegistry) public {
+        debtKernelAddress = _debtKernel;
         debtRegistry = DebtRegistry(_debtRegistry);
+        tokenRegistry = TokenRegistry(_tokenRegistry);
     }
 
-    function collateralize(
-        bytes32 agreementID,
-        address token,
-        uint amount,
-        uint lockupPeriodEndTimestamp
-    )
+    /**
+     * Abstract interface for a function that returns the last timestamp
+     * at which any repayment is expected -- i.e., the debt agreement's
+     * term's ending.
+     *
+     * @param agreementId bytes32 The debt agreement's ID
+     * @return _timestamp uint
+     */
+    function getTermEndTimestamp(
+        bytes32 agreementId
+    ) public view returns (uint _timestamp);
+
+    /**
+     * Hook that is called by the DebtKernel upon a debt agreement's
+     * issuance.  Parses out collateralization-specific parameters
+     * from the debt agreement's terms contract parameters and
+     * pulls collateral from the debtor based on those parameters.
+     *
+     * @param agreementId bytes32 The debt's agreement ID
+     * @param collateralizer address The debt agreement's collateralizer
+     * @return _success bool
+     */
+    function registerTermStart(bytes32 agreementId, address collateralizer)
         public
+        returns (bool _success)
     {
-        // the amount being put up for collateral must exceed 0.
-        require(amount > 0);
+        require(msg.sender == debtKernelAddress);
+
+        address collateralToken;
+        uint collateralAmount;
+        uint gracePeriodInDays;
 
-        // the lockup period must occur at some point in the future.
-        require(lockupPeriodEndTimestamp > block.timestamp);
+        (collateralToken, collateralAmount, gracePeriodInDays) = retrieveCollateralParameters(agreementId);
+
+        // the amount being put up for collateral must exceed 0.
+        require(collateralAmount > 0);
 
         /*
         Ensure that the agreement has not already been collateralized.
 
         If the agreement has already been collateralized, this check will fail
-        because any valid form of collateral will have a non-zero lockupPeriod.
-        Only an uncollateralized agreement would meet this requirement.
+        because any valid collateralization must have some sort of valid
+        address associated with it as a collateralizer.  Given that it is impossible
+        to send transactions from address 0x0, this check will only fail
+        when the agreement is already collateralized.
         */
-        require(collateralForAgreementID[agreementID].lockupPeriod == 0);
+        require(agreementToCollateralizer[agreementId] == address(0));
 
-        ERC20 erc20token = ERC20(token);
-        address collateralizer = msg.sender;
+        ERC20 erc20token = ERC20(collateralToken);
```

Comment:
Any exceptions to look out for here?

Reply:
To my knowledge, no -- this will still succeed if `collateralToken` is the null address.  We could assert that the address is not null before, but that would throw no less of an indecipherable error.

Worth revisiting [once `revert` with reason exists](https://github.com/ethereum/solidity/issues/1686).

Reply:
Come to think of it, actually, I think we should check for it. Would make debugging errors related to that easier, given that we'd have a line to comment / uncomment in order to check whether that invariant would be satisfied. Fixing.